### ISSUE TITLE
Team Call #3 - Jul 12 2021 @ 1830 EDT

### ISSUE THREAD

**Issue by gpthio** [MADS-Capstone-BLS-Job-Descriptions/predicting_demographics#5]
**Attendees**
- AF, SN, GT  
- Zoom call with Elle O'Brien (EO) followed by team Slack call
 
**Summary of call with EO**
- Suggested to build simple prototype then expand (e.g. text only + 1 binary % male) then build in features and try to predict more outputs 
- Lots of research on multiple feature models and multi-modal models, but break it down into bite-sized pieces
- First reaction was to use “hugging-face” NLP from French open-source group, state-of-the-art transformer models, in nice function wrappers for easy-input/output but then thought since using imagery and text, Keras would be better
- Keras: Good for building blocks, seems like easiest way without having to go into TF pr PyTorch
- Look at [Arxiv](https://arxiv.org/) for established combinations or just do trial-and-error since we are dealing with _Big Data_ rather than _Huge Data_
- Just treat each model separately, then combine, see how they do together. Based on last check in Keras, VGG can produce a feature vector --  e.g. VGG16 until last SoftMax layer, then just have a vector. NN for text is going to do something analogous -- e.g. give you 'mystery numbers' then append to last layer for classifications, pretty sure there are tutorials on how to add-in 
- Also have a 'dumb idea' ... can feed images to an (off-the-shelf) image recognition model to generate text caption
- Confirmed that a Neural Network (NN) is the right model approach given our multi-modal data and especially for test. Using any other models in this case would be sacrificing so much power given all the advances in NN, but recommend breaking into chunks for easier debugging
- Responded to AF question on interpretability by suggesting we look at LIME ([Local Interpretable Model-Agnostic Explanations](https://homes.cs.washington.edu/~marcotcr/blog/lime/))
- EO experience with Census data is that it is a matrix which you could melt into one long vector, maybe combine age bins
References

** Summary of follow-up team call**
- Agree to split up the problem and start with simple models
- Reconfirmed that we are doing a regression to predict specific levels of output variables (i.e. sex, age, race)
- Can’t do full model with the **Data for Occupations Not Covered in Detail (DONCD)** dataset 
- Sex: 1 level (% of women)
- Age: 7 levels (age buckets)
- Race: 5 levels (Black, Hispanic/Latino, White, Asian, Other). These will add up to > 100% due to Hispanic/Latino not being a racial category so they can and do identify as Black or White for example
- Job descriptions between the main OOH jobs and the CPS appear to match up, but not all headings are present in all job descriptions
<img width="596" alt="ooh_jd_sample" src="https://user-images.githubusercontent.com/75413805/125893446-58b084ef-58b4-49b0-8bdb-66371024f997.png">
- SN: Need to do a little more work to extract pay and education
- Agreed for initial simple models to extract all raw text from job descriptions rather than attempt to extract top down based on category headings 
- AF: Would be helpful to get the numerical data from DONCD -- i.e. separate fields for median wage, entry level education, 2019 employment, growth rate. AF: This may be tough since some of this data is textual and.or categorical with terms such as "moderate"
- AF: Although DONCD has neither images nor text, intend to use this more as a check to see if it yields significantly different results

**References**
[arXiv.org](https://arxiv.org/)
[Local Interpretable Model-Agnostic Explanations](https://homes.cs.washington.edu/~marcotcr/blog/lime/)
[SHAP: How to Interpret Machine Learning Models With Python
Explainable machine learning with a single function call](https://towardsdatascience.com/shap-how-to-interpret-machine-learning-models-with-python-2323f5af4be9)

**Tasks**
- [ ] ALL: Get familiarized with Keras
- [ ] SN: Update OOH text file
- [ ] GT: Find and run a good image captioning model to generate text descriptions of OOH job description images
